Fix MPI communicator for serial builds

julianakwan opened the pull request:
After AMRex [PR#4463](https://github.com/AMReX-Codes/amrex/pull/4463), `MPI_COMM_WORLD` is no longer defined for serial builds. This is breaking our CI because we test for both `USE_MPI=TRUE` and `USE_MPI=FALSE`. 

I have removed the MPI communicator from calls to `amrex::Initialize` since we never define our own MPI communicator and it should default to `MPI_COMM_WORLD` in this case. 

The only exception is `SetupFunctions.hpp` in which I have defined a pre-processor guard for serial builds so that `MPI_COMM_WORLD` is recognized as `amrex::MPI_COMM_WORLD` (as per Weiqun's suggestion). This is because we were passing the other argument to `ParmParse` to avoid using the managed memory Arena. This should be the default behaviour anyway, but I think someone should look into it before I remove it 😄

julianakwan committed:
Core: Update MPI communicator to use AMReX namespace

AMReX PR#4463 wrapped the MPI communicators in a AMReX namespace
which would evaluate to nothing for non-MPI builds. This broke our
CI when USE_MPI=FALSE as MPI_COMM_WORLD no longer exists but
amrex::MPI_COMM_WORLD does.

julianakwan committed:
Core: Remove MPI communicator from amrex:Initialize

This should remove the need for an additional macro to wrap
up MPI_COMM_WORLD as we don't use our own communicator
anyway. For the main GRTeclyn amrex::Initialize, I added a
preprocessor guard for non-MPI builds so that the definition
of MPI_COMM_WORLD in the amrex namespace is used instead.

mirenradia reviewed (changes requested):
Since `amrex.the_arena_is_managed = false` is now the default in AMReX (since AMReX-Codes/amrex#3438 in Jul 2023), I think we can safely remove the lambda that we pass to `amrex::Initialize` in `SetupFunctions.hpp` and just pass the `argc` and `argv` arguments.

julianakwan committed:
Core: Remove parm parse from amrex::Initialize

This will remove the boolean argument passed to parm parse
as per Miren's request for PR#120.

mirenradia reviewed (approved):
LGTM!
